onnx/defs/math/defs.cc

```diff
@@ -505,24 +505,48 @@ ONNX_OPERATOR_SET_SCHEMA(
             "Constrain input and output types to float tensors.")
         .TypeAndShapeInferenceFunction(propagateShapeAndTypeFromFirstInput));
 
-static const char* Pow_ver7_doc = R"DOC(
+static const char* Pow_ver12_doc = R"DOC(
 Pow takes input data (Tensor<T>) and exponent Tensor, and
 produces one output data (Tensor<T>) where the function `f(x) = x^exponent`,
 is applied to the data tensor elementwise.
 )DOC";
 
 ONNX_OPERATOR_SET_SCHEMA(
     Pow,
-    7,
+    12,
     OpSchema()
-        .SetDoc(std::string(Pow_ver7_doc) + GenerateBroadcastingDocMul())
+        .SetDoc(std::string(Pow_ver12_doc) + GenerateBroadcastingDocMul())
         .Input(0, "X", "First operand, base of the exponent.", "T")
-        .Input(1, "Y", "Second operand, power of the exponent.", "T")
+        .Input(1, "Y", "Second operand, power of the exponent.", "T1")
         .Output(0, "Z", "Output tensor (same size as X)", "T")
         .TypeConstraint(
             "T",
-            {"tensor(float16)", "tensor(float)", "tensor(double)"},
-            "Constrain input and output types to float tensors.")
+            {"tensor(uint8)",
```

Comment:
I don't think it's a good idea to have 8-bit integer support added as this, the output will overflow if it has same type as input.

What's the motivation of adding integer support right now? can adding int32/int64 support be good enough? if yes, don't add 8/16 bit integers for now.

Reply:
that's a good point, thanks.
this is needed for the function ONNX::MeanSquaredDistance that executes a Pow operation with X as a float and Y as an int.
I added ints for X as well for completeness since it is a valid operation that we would need eventually. I removed 8-16 bits types for T (X and output).